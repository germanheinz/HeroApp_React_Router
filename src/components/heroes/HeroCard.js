import React from 'react';
import { Link } from 'react-router-dom';

export const HeroCard = ({ id, superhero, publisher, alter_ego, first_appearance, characters }) => {

    return (
        <div className="card ms-3" style={{ maxWidth: 200 }}>
            <div className="card-body">
                <h5 className="card-title">{ superhero }</h5>
                <h6 className="card-subtitle mb-2 text-muted">{ alter_ego }</h6>
                <img src={`../assets/heroes/${ id }.jpg`} alt="hero" className="card-img"/>
                <p className="card-text">the card's content.</p>

                <Link to={`./hero/${ id }`}> More.. </Link>
                
            </div>
        </div>
    )
}
