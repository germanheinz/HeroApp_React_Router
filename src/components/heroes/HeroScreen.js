import React, { useMemo } from 'react';
import { Redirect, useParams } from 'react-router';
import { getHeroById } from '../selectors/getHeroById';

export const HeroScreen = ({ history }) => {

    const { heroId } = useParams();
    console.log( heroId );

    // const hero = getHeroById( heroId );
    const hero = useMemo(() => getHeroById( heroId ), [ heroId ])

    console.log('Hero', hero, 'HeroId', heroId);

    console.log(hero);

    if(!hero){
        return <Redirect to="/" />
    }
    
    const { 
        alter_ego,
        characters,
        first_appearance,
        id,
        publisher,
        superhero
    } = hero;
    
    console.log( id );

    const handleReturn = () => {

        if(history.length <=2){
            history.push('/');
        }else{
            history.goBack();
        }
    }
    
    return (
        <div>
            <li>
                <div className="card ms-3" style={{ maxWidth: 200 }}>
                    <div className="card-body">
                            <h5 className="card-title">{ superhero }</h5>
                            <h6 className="card-subtitle mb-2 text-muted">{ alter_ego }</h6>
                            <img src={`../assets/heroes/${ id }.jpg`} alt="hero" className="card-img"/>
                            <p className="card-text">{ first_appearance }</p>
                    </div>
                    <button className="btn btn-outline-info" onClick={ handleReturn }>Back</button>
                </div>
            </li>
        </div>
    )
}
