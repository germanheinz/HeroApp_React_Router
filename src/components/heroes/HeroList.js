import React, { useMemo } from 'react';
import { getHeroesByPublisher } from '../selectors/getHeroesByPublisher';
import { HeroCard } from './HeroCard';


export const HeroList = ({ publisher }) => {

    const heroesMemo = useMemo(() => getHeroesByPublisher( publisher ), [ publisher ]);
    // const heroes = getHeroesByPublisher( publisher );

    return (
        <>
        {
            heroesMemo.map( hero => (
                <HeroCard 
                    key={ hero.id }
                        { ...hero }
                />
            ))
        }
        </>
       
    )
}
