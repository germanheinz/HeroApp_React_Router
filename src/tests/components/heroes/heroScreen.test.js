import React from 'react';
import { mount } from 'enzyme';
import { HeroScreen } from '../../../components/heroes/HeroScreen';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router';

 
Enzyme.configure({ adapter: new Adapter() });


describe('Testing en <HeroScreen />', () => {
    
    const history = {
        length: 10,
        push: jest.fn(),
        goBack: jest.fn()
    }

    const wrapper = mount(
        <MemoryRouter>
            <HeroScreen history={ history }/>
        </MemoryRouter>
    );

    test('Debe de mostrar el componente redirect si no hay argumentos en el URL ', () => {
        expect(wrapper.find('Redirect').exists()).toBe(true);
    })
    

})
