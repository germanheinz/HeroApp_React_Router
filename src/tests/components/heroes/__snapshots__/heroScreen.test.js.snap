// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing en <HeroScreen /> Debe de mostrar el componente redirect si no hay argumentos en el URL  1`] = `ReactWrapper {}`;

exports[`Testing en <HeroScreen /> Debe mostrarse correctamente  1`] = `ReactWrapper {}`;
